File: Projects/07/Names.java
/*
 * Name: Emily Costa
 * Date: 4/8/2020
 * Proejct 7
 ***************************/

import java.util.Scanner;

class Name {

	private String myFirst, myMiddle, myLast;

	public Name (String first, String middle, String last){
		myFirst = first;
		myMiddle = middle;
		myLast = last;
	}

	// function to change first name
	void changeFirst(String newName) {
		myFirst = newName;
	}

	// function to change middle name
	void changeMiddle(String newName) {
		myMiddle = newName;
	}

	// function to change last name
	void changeLast(String newName) {
		myLast = newName;
	}

	//operation to return last, frist, middle initial
	public String lfmi() {
		return myLast + ' ' + myFirst + ' ' + myMiddle.charAt(0);
	}

	// fuction to print name
	public String toString(){
    	return myFirst + ' ' + myMiddle + ' ' + myLast;
    } 

};

public class Names {


	public static Name Initialize() {

		// prompt user for names and scan them in
		System.out.print("\n Enter first name: ");
		Scanner keyboard = new Scanner(System.in);
		String first = keyboard.nextLine();
		System.out.print(" Enter middle name: ");
		String middle = keyboard.nextLine();
		System.out.print(" Enter last name: ");
		String last = keyboard.nextLine();

		Name aName = new Name(first, middle, last);

		return aName;
	} 

	public static void main (String[] args){
		// call fucntion to intiialize new Name
		Name aName = Initialize();
		// print name
		System.out.println(" " + aName);
		// print last, first, middle initial
		System.out.print(" " + aName.lfmi());

		// Change first name
		System.out.print("\n\n Enter new first name: ");
		Scanner keyboard = new Scanner(System.in);
		String name = keyboard.nextLine();
		aName.changeFirst(name);
		System.out.println(" " + aName);

		// Change middle Name
		System.out.print("\n Enter new middle name: ");
		name = keyboard.nextLine();
		aName.changeMiddle(name);
		System.out.println(" " + aName);

		// Change last Name
		System.out.print("\n Enter new last name: ");
		name = keyboard.nextLine();
		aName.changeLast(name);
		System.out.println(" " + aName + "\n");
		// print last, first, middle intitial of new name
		System.out.print(" " + aName.lfmi() + "\n\n");

	}

}

File: Projects/07/names.clj

;;; Emily Costa
;;; 4/8/2020
;;; Project 7
;;;;;;;;;;;;;;;;;;;;;;;

(ns names)

; Definiion of record-type Name
(defrecord Name [firstName middleName lastName])

; function to initialize Name
(defn make-Name []
	(printf "\nEnter first name: ") (flush)
	(let [
		first (read-line) ]

		(printf "Enter middle name: ") (flush)
		(let [
	 	middle (read-line)] 

	 	(printf "Enter last name: ") (flush)
	 	(let [
	 		last (read-line)]

	 		(->Name first middle last)

	 	) ; end let
	 ) ; end let
	) ; end let
) ; end initialize-Name

(defn getFirst [^Name aName]
		(:firstName aName)
)

(defn getMiddle [^Name aName]
		(:middleName aName)
)

(defn getLast [^Name aName]
		(:lastName aName)
)

(defn toString [^Name aName]
	(str (getFirst aName) " " (getMiddle aName) " " (getLast aName))
)

(defn printName [^Name aName]
	(print (toString aName))
)

; function to change first name
(defn changeFirst [^Name aName]
	(printf "\nEnter new first name: ") (flush)
	(let [
		name (read-line)]
		(assoc aName :firstName name)
	)
)

; function to change middle name
(defn changeMiddle [^Name aName]
	(printf "Enter new middle name: ") (flush)
	(let [
		name (read-line)]
		(assoc aName :middleName name)
	)
)

; function to change last name
(defn changeLast [^Name aName]
	(printf "Enter new last name: ") (flush)
	(let [
		name (read-line)]
		(assoc aName :lastName name)
	)
)

; function to return last, first, middle initial 
(defn lfmi [^Name aName]
	(str (getLast aName) " " (getFirst aName) " " (subs (getMiddle aName) 0 1))
)

(defn -main []

	(let [
			aName (make-Name) ]

	(printName aName) (println)
	(printf (lfmi aName)) (println)

	(let [
		aName (changeFirst aName)
		aName (changeMiddle aName)
		aName (changeLast aName) ]

	(printName aName) (println)
	(printf (lfmi aName)) (println) (println)

	) ; end let
	) ; end let
)